test/integration/test.macros.js

```diff
@@ -327,3 +327,142 @@ QUnit.test('tcfMacros', function(assert) {
 
   window.__tcfapi = oldtcf;
 });
+
+QUnit.test('default macros should not be replaced when disableDefaultMacros is true', function(assert) {
+  const string = '{player.id}';
+  const customMacros = {
+    disableDefaultMacros: true
```

Comment:
suggestion/non-blocking: I would suggest that we have a test that ensures that "magic" properties of `customMacros` do not get replaced themselves. Something like:
```js
const string = 'disableDefaultMacros';
const result = this.player.ads.adMacroReplacement(string, false, {
  disableDefaultMacros: false
});

assert.strictEqual(result, string, 'disableDefaultMacros was not treated as a macro');
```